fixed the checkJointTypes function for the case of TORQUE control

AlexAntn opened the pull request:
This PR fixes issue #718 .

Since we have an array of PidInfo / TrqPidInfo, we can't rely on class inheritance to query the parameters of each PID ([an array of Derived is not a kind-of array of Base][1]). As such, we need to check if the PID type is torque and handle it differently.

[1]: https://www.cs.technion.ac.il/users/yechiel/c++-faq/array-derived-vs-base.html

pattacini reviewed (changes requested):
I can see a verbatim copy'n'paste of lots of code lines we could thus refactor and made much more compact.

pattacini commented:
@AlexAntn we can get away w/ less by refactoring the main routine and make it call 2 sub-routines that implement each the body of the 2 for loops.

AlexAntn commented:
Ìn the last commit to this issue I have refactored the loops in order to reduce duplicated code. I have also changed how we obtain the `firstjoint` in order to have a single loop.

These changes have been tested directly on the robot and are working (`yarprobotinterface` started with no issues)

pattacini commented:
LGTM, thanks a lot @AlexAntn 🚀

pattacini commented:
I'll wait for the CI to complete and the squash and merge the PR.

@Nicogene in the meantime, you may be willing to review it as well.

Nicogene reviewed (approved):
LGTM

Nicogene commented:
I think we can squash all the commit in just one

pattacini commented:
>I think we can squash all the commit in just one

Don't do that please, otherwise CI will start over from scratch!

I can simply use the corresponding button from the GH GUI instead.
